### ISSUE TITLE
API users want a messageID API that works for all topics / datatypes

### ISSUE THREAD

**Issue by jimduff-usds** [CDCgov/prime-reportstream#8033]
The current messageID API only works for covid.  This ticket is to make it work for all future datatypes, by using the new ITEM table, created in #8032.

This ticket includes some tech debt cleanup on the messageID API as well.
#### REST API Name change
The concept of `message` is HL7 specific.   Generalize to the term `item` which can be a message, a row, or a bundle, as follows:
- Change `api/messages/search?messageID=xxx` to `api/items?trackingId=xxx`.
- Change `api/message/{covid_results_metadata_id}` to `api/item/{item_id}` 
- Fix the documentation too (docs/api/message.yml, release notes, kotlin code).
- Rename Message to something else (probably not Item, as that'll be generated by Jooq).  Maybe ItemHistory?

#### Search on item_id, not covid_results_metadata_id

- The "id' returned in the new ItemHistory object, (returned by the api/items endpoint) should now always be the item_id, the primary key of the item table.
- For each item found by any query in the ITEM table, also return its metadata. 

#### Other improvements
- Add 'topic' to the ItemHistory object so we know which metadata table it came from.  THis is found in the ReportFile table.

#### Tech debt cleanup:
- The .kt files for this need to move to src/kotlin/history/azure.  messageID queries are history API queries, so they properly belong in history/azure.
- ItemHistory (was Message) should be its own class to parallel DeliveryHistory.kt etc.

### Implementation Notes

The linking of ITEM  - via Topic - to the corresponding ***_Metadata table can be accomplished in several different ways.   
- returns just the generic ITEM data, then requiring a second, topic-specific query, to get the *_metadata.
- using a fancy table inheritance design.    If queries are going to go across topics, or return a large number of ITEMs, this is the most performant solution.  However, I think its its own separate ticket.
- Require the user to send a topic parameter.   this is a nice MVP solution.  Use the topic to determine which metadata table to join with.     A query can only get data for one topic.
-

**Comment by avnieldravid** [comment]
Waiting on Uinversal Search design to create Epic.
